### ISSUE TITLE
bug: <WasmEdge/example/plugin/get-string can not be compiled>

### ISSUE THREAD

**Issue by PinappleUnderTheSea** [WasmEdge/WasmEdge#3316]
### Summary

Both when I try to build and run get-string plugin by cmake and running .wasm file , I will meet some error and `test.wasm` can not be compiled

### Reproduction steps

1. If I run `mkdir build && cd build && cmake ..`, the terminal returns :
```
CMake Error at CMakeLists.txt:14 (target_include_directories):
  Error evaluating generator expression:

    $<TARGET_PROPERTY:wasmedgePlugin,INCLUDE_DIRECTORIES>

  Target "wasmedgePlugin" not found.

```

2. If I run `wat2wasm test.wat -o test.wasm && wasmedge test.wasm`, the error message is :
```
[2024-04-02 15:20:42.821] [error] instantiation failed: unknown import, Code: 0x302
[2024-04-02 15:20:42.822] [error]     When linking module: "module_wasm_name" , function name: "get_string"
[2024-04-02 15:20:42.822] [error]     At AST node: import description
[2024-04-02 15:20:42.822] [error]     At AST node: import section
[2024-04-02 15:20:42.823] [error]     At AST node: module

```

I asked this question in Discord. The admin helped me a lot and suggest me to raise an issue here.

### Screenshots

![image](https://github.com/WasmEdge/WasmEdge/assets/109454651/ef775273-3e3c-4fbb-852f-9179dfb35fce)
![image](https://github.com/WasmEdge/WasmEdge/assets/109454651/60660c7a-c522-4cf7-ba00-5c4a13d34bfc)

### Components

CLI

### WasmEdge Version or Commit you used

commit ecdee81f71bd4ae980df609ebff7ac1a7bc2148e 

### Operating system information

Ubuntu 22.094

### Hardware Architecture

x86_64

**Comment by hydai** [comment]
@q82419 
Could you please check this? Should we use the wasi-logging as the example instead of this one?

**Comment by hydai** [comment]
The build command should be:

```
cd WasmEdge
cmake -Bbuild -DWASMEDGE_BUILD_EXAMPLE=ON .
cmake --build build
```

And we will encounter another error due to the plugin API changes:

```
/Users/hydai/workspace/WasmEdge/examples/plugin/get-string/getstring.cpp:97:9: error: no type named 'PluginRegister' in namespace 'WasmEdge::Plugin'
Plugin::PluginRegister Register(&Descriptor);
~~~~~~~~^
1 error generated.
```

**Comment by PinappleUnderTheSea** [comment]
sorry maybe i have no idea about it because i am a freshman haha. I am just using and modifying this example to test my hostfuntion.

**Comment by q82419** [comment]
Fix this with PR #3322.
